### ISSUE TITLE
Printing in ISR

### ISSUE THREAD

**Issue by wizard97** [charlescao460/iSkipper-Software#1]
I quickly looked at your Arduino sketch and noticed that you are printing over Serial in your packet handlers. The packet handlers run in an ISR, and printing over Serial in an ISR is undefined behavior.

**Comment by charlescao460** [comment]
Yeah actually I've found that there are some wired things happens to  my Arduino Sketch when replacing answers. But it seems to be normal when capturing and sending. Anyway, I am going to refactor  the entire sketch. As you might noticed, current sketch is messy and hard to modify _(:зゝ∠)_. I'll let you know when I finish refactor.
Thank you very much  for this  notice.  I didn't know this before.

**Comment by charlescao460** [comment]
I've re-written the entire sketch in #5. It should work fine.
